### PR TITLE
New mochify

### PR DESCRIPTION
**Pull request by mantoni**
#### Purpose (TL;DR) - mandatory

I'm working on a rewrite of mochify which aims to solve #380, is more modular and works with any bundler.

#### Background (Problem in detail)  - optional

The current mochify implementation is dated and does not support the latest webdriver protocols.

#### Solution  - optional

The new mochify implementation works with pluggable drivers. The `@mochify/driver-webdriver` uses the official `webdriver` module internally.

~~Interestingly, this successfully runs tests in Safari (although there are failing tests), but I'm having trouble running tests in Firefox and Edge.~~

New Mochify development happens on the `rewrite` branch here: https://github.com/mantoni/mochify.js/tree/rewrite

### CODE REVIEW

**Comment by fatso83** [comment]
Only thing that is failing is the missing Chromium:
```
Run export SINON_CHROME_BIN=$(which google-chrome-stable)

> @sinonjs/fake-timers@7.1.2 test-headless
> mochify --driver puppeteer "--chrome" "/usr/bin/google-chrome-stable" "--allow-chrome-as-root"

Error: Could not find expected browser (chrome) locally. Run `npm install` to download the correct Chromium revision (901912).
```

**Comment by mantoni** [comment]
Builds now pass, except for Safari which has failing unit tests.

**Comment by fatso83** [comment]
Any idea what the "kl" in the output is? Some debugging statement left in?
```
> @sinonjs/fake-timers@8.0.0 test-headless /Users/carlerik/dev/fake-timers
> mochify --driver puppeteer

kl

  .................................,,,,,,,,,,,,,,,.........................................................................................................
  .............................................,...,...........,...........................,,...................................,..............,......,,,,,
  ,,,,,,...,,,,,,,,,,,,,,,,,..........,..,,........,,..

  304 passing (5s)
  55 pending
```

**Comment by mantoni** [comment]
> Any idea what the "kl" in the output is?

When I run it locally with the current state of mochify linked locally, this doesn't appear.

**Comment by fatso83** [comment]
Hah, probably me that has fiddled around with node_modules when debugging at some earlier stage and forgot 😄

**Comment by fatso83** [comment]
Just wondering about the tests that fail in Safari ... how do I run these Mochify tests locally against a local Safari instance?

**Comment by mantoni** [comment]
You can launch `safaridriver` locally. It ships with Safari.

Setup and config examples are documented here: https://github.com/mantoni/mochify.js/blob/rewrite/driver-webdriver/README.md

**Comment by mantoni** [comment]
I finally manager to get a new Mochify release out and updated this PR.

I have no idea why the selenium job is skipped. Anyone?

Happy Christmas 🎄

File: mochify.webdriver.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+"use strict";
```
Comment (SimenB) [inline mochify.webdriver.js:1]: we should ignore this file when collecting code coverage

**Comment by mantoni** [comment]
After a small adjustment, tests now also pass in all browsers, including Safari. Unfortunately Safari does not include the function name in the `err.stack`, so I had to make some assertions less specific.

While testing I also removed Browserify and used ESBuild instead. If this is not wanted, I'm happy to switch it back.

**Review by SimenB — APPROVED** [review]

**Review by mroderick — APPROVED** [review]
Very nice 👍

**Pull request merged**